### ISSUE TITLE
Neural Machine Translation with Word Predictions

### ISSUE THREAD

**Issue by jlibovicky** [ufal/neuralmonkey#514]
Paper _Neural Machine Translation with Word Predictions_ by Weng et al. (https://arxiv.org/pdf/1708.01771.pdf) improves the decoder training by adding an additional objective by predicting the remaining target sentence words from the decoders' states. In this way there were able improve by 4.5 BLEU points in Chinese-English translation and 1.3 BLEU points in German-English translation. Similar trick helped with encoder.

This can be added to Neural Monkey by implementing a new decoder operating on states of the recurrent decoder/encoder and do multi-task training in the standard way.

**Comment by guotong1988** [comment]
Have you finished the implemention? Thank you!

**Comment by jindrahelcl** [comment]
No, nor we are planning to right now. You can contribute if you want :)

**Comment by guotong1988** [comment]
I confuse about the paper: the author want to predict the vocab OR just another additional objective designed by the author?
